### ISSUE TITLE
Question about codelite-ctags

### ISSUE THREAD

**Issue by uffejakobsen** [eranif/codelite#2929]
Hi, 
I see that the new CL-15.0.9 sources carries precompiled binary versions of what looks to be stock versions of universal-ctags version 5.9

I'm maintaining two codelite AUR packages for Archlinux (codelite(-stable) and codelite-unstable)

I know for sure that some users would not like the fact that they just have to trust such a supplied precompiled binary without being able to inspect/build from source

But as Archlinux already has universal-ctags as a package - my plan was simply to create /usr/bin/codelite-ctags as a symlink to /usr/bin/ctags

Looking at the ctags optional feature list (ctags --version) the Archlinux ctags version has all features of codelite-ctags (and more) 

codelite-ctags: Optional compiled features: +wildcards, +regex, +gnulib_regex, +iconv, +option-directory, +json, +interactive, +sandbox, +packcc, +optscript, +pcre2

universal-ctags: Optional compiled features: +wildcards, +regex, +gnulib_regex, +iconv, +option-directory, +xpath, +json, +interactive, +sandbox, +yaml, +packcc, +optscript

Now my question is if your precompiled version of universal-ctags is modified in any way - eg. carrying extra functionality created by you ? so that my plan would not work ?

br/Uffe :-)

**Comment by eranif** [comment]
I had some problems with Linux (not all of them ofc)
On Ubuntu 18.04 - there is no such package, while on 20.04 - there is
On Amazon Linux 2 - no universal ctags
You see where I am going with it...

The only solution I could think of (for now) was to build the executable as non dynamic executable (static linking libgcc etc) so the same binary can run on basically all x86_64 linux systems (which it does).

Basically, if you can simply create a symlink from /usr/bin/codelite-ctags -> ctags (your distro ctags) it will work.
You can test it locally by:

- open the file `WORKSPACE/.ctagsd/ctagsd.json`
- replace the value under `"codelite_indexr:"` to point to your `ctags` and restart `ctagsd` process (just kill it, CodeLite will spawn another one)
-

**Comment by eranif** [comment]
FYI: the version of ctags that comes with CodeLite includes a patch that is not yet applied to the official u-ctags.
I assume it will be applied soon

**Comment by uffejakobsen** [comment]
Thanks - yes, I noticed your commit about patched u-ctags

I do not know what the patch is about - from the commit text it seems to be related to something about macros 
But I guess that an unpatched u-ctags will just return fewer or no hits for queries on macros ?

**Comment by eranif** [comment]
Yes, it will still work. But it wont be able to extract "recursive" macros previously defined as an empty string
